pkg/api/v1/query.go

```diff
@@ -0,0 +1,58 @@
+package v1
+
+import (
+	"github.com/els0r/goProbe/pkg/api/json"
+
+	"fmt"
+	"net/http"
+
+	"github.com/els0r/goProbe/pkg/query"
+)
+
+// handleQuery can be used to query the flow database that goPorbe writes
+// to disk. It is the equivalent of calling the binary `goQuery` on
+// the machine running goProbe.
+func (a *API) handleQuery(w http.ResponseWriter, r *http.Request) {
+	// set up default options
+	callerString := fmt.Sprintf("goProbe-API/%s", a.Version())
+	opts := []query.Option{
+		query.WithFormat("json"),
+	}
+
+	// create bare query arguments
+	args := query.NewArgs("", "", opts...)
+
+	// parse additional arguments from command line
+	err := json.Parse(r, &args)
+	if err != nil {
+		status := http.StatusBadRequest
```

Comment:
Minor nitpick: IMHO these three lines could / should be merged into a neat multi-line return in the style (same further below):
```
http.Error(w, fmt.Sprintf(...),
http.StatusBadRequest)

Allocating a temporary variable just to copy a const seems less readable and unneccessary...